geopandas/sindex.py

```diff
@@ -89,6 +99,22 @@ def __init__(self, geometry):
                 [None] * self.geometries.size, dtype=object
             )
 
+        @property
+        def valid_query_predicates(self):
+            # TODO: can we dynamically query these from somewhere?
+            # TODO: what about different Shapely versions?
+            return {
+                None,
+                "intersects",
+                "within",
+                "contains",
+                "overlaps",
+                "crosses",
+                "touches",
+                "covers",
+                "contains_properly",
```

Comment:
Added these two new predicates, but I haven't added any tests for these specific predicates.

Reply:
Ideally we would add some tests yes, if we are adding new allowed predicates (for the rtree version, it is our own code that needs to handle this, and not just passing it through to pygeos. Eg for "contains_properly", you will need to add that to the check whether to use prepared geometry or not)

Reply:
You are right, I forgot that _we_ implemented the predicate logic in rtree. I'll add a couple of tests.

Reply:
Ok I implemented some tests.

I am still of the opinion that it'd be worth it to have a naïve sindex implementation that is just two nested for loops checking each predicate. That way we can make a test set of 10,000 geometries or something and run through them all (these tests are fast). We'd also be able to accomodate each spatial index's predicates (i.e. `pytest.mark.parametrize("predicate", list(sindex.get_sindex_class().valid_query_predicates))`)